addons/binding/org.openhab.binding.globalcache/src/main/java/org/openhab/binding/globalcache/discovery/GlobalCacheDiscoveryService.java

```diff
@@ -0,0 +1,180 @@
+ */
+package org.openhab.binding.globalcache.discovery;
+
+import static org.openhab.binding.globalcache.GlobalCacheBindingConstants.*;
+
+import java.io.IOException;
+import java.net.SocketException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.config.discovery.DiscoveryServiceCallback;
+import org.eclipse.smarthome.config.discovery.ExtendedDiscoveryService;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link GlobalCacheDiscoveryService} class implements a service
+ * for discovering the GlobalCache devices.
+ *
+ * @author Mark Hilbush - Initial contribution
+ */
+public class GlobalCacheDiscoveryService extends AbstractDiscoveryService implements ExtendedDiscoveryService {
+    private final static Logger logger = LoggerFactory.getLogger(GlobalCacheDiscoveryService.class);
+
+    private ScheduledFuture<?> gcDiscoveryJob;
+
+    // Discovery parameters
+    public final static boolean BACKGROUND_DISCOVERY_ENABLED = true;
+    public final static int BACKGROUND_DISCOVERY_DELAY = 10;
+
+    private DiscoveryServiceCallback discoveryServiceCallback;
+
+    private boolean terminate;
+
+    private Runnable discoverRunnable = new Runnable() {
+        @Override
+        public void run() {
+            discover();
+        }
+    };
+
+    public GlobalCacheDiscoveryService() {
+        super(SUPPORTED_THING_TYPES_UIDS, 0, BACKGROUND_DISCOVERY_ENABLED);
+        gcDiscoveryJob = null;
+        terminate = false;
+    }
+
+    @Override
+    public void setDiscoveryServiceCallback(DiscoveryServiceCallback dscb) {
+        discoveryServiceCallback = dscb;
+    }
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypes() {
+        return SUPPORTED_THING_TYPES_UIDS;
+    }
+
+    @Override
+    protected void activate(Map<String, Object> configProperties) {
+        logger.info("Globalcache discovery service activated");
+        super.activate(configProperties);
+    }
+
+    @Override
+    protected void deactivate() {
+        logger.info("Globalcache discovery service deactivated");
+        stopBackgroundDiscovery();
+        super.deactivate();
+    }
+
+    @Override
+    protected void startBackgroundDiscovery() {
+        if (gcDiscoveryJob == null) {
+            terminate = false;
+            logger.info("Starting background discovery job in {} seconds", BACKGROUND_DISCOVERY_DELAY);
+            gcDiscoveryJob = scheduler.schedule(discoverRunnable, BACKGROUND_DISCOVERY_DELAY, TimeUnit.SECONDS);
```

Comment:
As you permanently require a thread for your own, it does not make sense to take this from the thread pool as the pool might run out of threads if everyone does it. So better spawn off your own thread for this purpose.